### ISSUE TITLE
pox contract: parameter in allow-contract-caller for upper limit does not work

### ISSUE THREAD

**Issue by friedger** [stacks-network/stacks-blockchain#2522]
**Describe the bug**
Calling `allow-contract-caller` with a `until-burn-ht` value in the future, results in stacks function to fail with permission denied.

**Steps To Reproduce**
1. deploy a contract that calls e.g. `delegate-stx`
1. make a `allow-contract-caller` call with a `until-burn-ht`
1. call the contract function that calls the genesis `delegate-stx` function
1. see the error `9` permission denied.

**Expected behavior**
The contract call should fail only after `until-burn-ht` block has passed.

Reporting it here as discovered with @yknl

**Comment by kantai** [comment]
Yep -- this appears to be broken in the contract-caller check in `pox.clar`:


```
          ;; is the caller allowance expired?
          (if (< burn-block-height (unwrap! (get until-burn-ht caller-allowed) true))
              false
              true))))
```

This should be `>`, not `<`. This will need to be fixed in Stacks 2.1, and `until-burn-ht` won't work correctly as a parameter to `allow-contract-caller` until then.

**Comment by jcnelson** [comment]
Implemented in #2755
